frontend/request.go

```diff
@@ -116,3 +119,74 @@ func marshalDockerfile(ctx context.Context, dt []byte, opts ...llb.ConstraintsOp
 	st := llb.Scratch().File(llb.Mkfile(dockerui.DefaultDockerfileName, 0600, dt), opts...)
 	return st.Marshal(ctx)
 }
+
+func ForwardToSigner(ctx context.Context, client gwclient.Client, platform *ocispecs.Platform, cfg *dalec.Frontend, s llb.State, filePattern string) (llb.State, error) {
+	const (
+		sourceKey  = "source"
+		contextKey = "context"
+		inputKey   = "input"
+
+		gatewayFrontend = "gateway.v0"
+	)
+
+	opts := client.BuildOpts().Opts
+	id := identity.NewID()
+
+	req, err := newSolveRequest(toFrontend(cfg))
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	for k, v := range opts {
+		if k == "source" || k == "cmdline" {
+			continue
+		}
+		req.FrontendOpt[k] = v
+	}
+
+	inputs, err := client.Inputs(ctx)
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	m := make(map[string]*pb.Definition)
+
+	for k, st := range inputs {
+		def, err := st.Marshal(ctx)
+		if err != nil {
+			return llb.Scratch(), err
+		}
+		m[k] = def.ToPB()
+	}
+	req.FrontendInputs = m
+
+	stateDef, err := s.Marshal(ctx)
+	if err != nil {
+		return llb.Scratch(), err
+	}
+
+	req.FrontendOpt[contextKey] = compound(inputKey, id)
```

Comment:
This should be the context.

Reply:
It is, isn't it? `contextKey`  is `"context"`

Reply:
Sorry, I think I flipped this in my head.
Looks like this is `opts["context"] = "input:<id>"`
However I think `id` should litterally be `context` by convention.